File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string txt);
int count_words(string txt);
int count_sentences(string txt);
int main(void)
{
    int letters, words, sentences, rounded_index;
    float index, l, s;
    string txt = get_string("Text: ");
    letters = count_letters(txt); //Counts letters.
    words = count_words(txt);// Counts words.
    sentences = count_sentences(txt); //Counts sentences.
    l = (float)letters / words * 100;
    s = (float)sentences / words * 100;
    index = 0.0588 * l - 0.296 * s - 15.8;//Calculates index using Coleman-Liau formula.
    rounded_index = round(index);
    if (rounded_index >= 16) //If rounded index is 16 ang higher.
    {
        printf("Grade 16+\n");
    }
    else if (rounded_index < 1) //If rounded index is below 1.
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", rounded_index);

    }
}








int count_letters(string txt)
{
    int count = 0;
    for (int x = 0; txt[x] != '\0'; x++)
    {
        if ((txt[x] >= 'A' && txt[x] <= 'Z') || (txt[x] >= 'a' && txt[x] <= 'z'))
        {
            count++;
        }
    }
    return count;

}

int count_words(string txt)
{
    int count = 1; //Starting from 1 because there is no ' ' character after the last word.
    for (int x = 0; txt[x] != '\0'; x++)
    {
        if (txt[x] == ' ')
        {
            count++;
        }
    }
    return count;

}



int count_sentences(string txt)
{
    int count = 0;
    for (int x = 0; txt[x] != '\0'; x++)
    {
        if (txt[x] == '.' || txt[x] == '!' || txt[x] == '?')
        {
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
import cs50


def count_letters(txt):
    count = 0
    x = 0
    for i in txt:
        if (txt[x] >= 'A' and txt[x] <= 'Z') or (txt[x] >= 'a' and txt[x] <= 'z'):
            count = count+1
        x = x+1
    return count


def count_words(txt):
    count = 1
    x = 0
    for i in txt:
        if txt[x] == ' ':
            count = count+1
        x = x+1
    return count


def count_sentences(txt):
    count = 0
    x = 0
    for i in txt:
        if txt[x] in ['.', '!', '?']:
            count = count+1
        x = x+1
    return count


def main():
    print("Text: ", end='')
    text = input()
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    #print(f"letters {letters}")
    #print(f"words {words}")
    #print(f"sentence {sentences}")
    l = letters / words * 100
    s = sentences / words * 100
    index = 0.0588 * l - 0.296 * s - 15.8
    #print(f"index {index}")
    rounded_index = round(index)

    if rounded_index >= 16:  # If rounded index is 16 ang higher.
        print("Grade 16+\n")
    elif rounded_index < 1:  # If rounded index is below 1.
        print("Before Grade 1\n")
    else:
        print(f"Grade {rounded_index}")


if __name__ == "__main__":
    main()